src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -218,35 +229,36 @@ public boolean hasGiverRespondedForSession(String userEmail, String feedbackSess
 
     public List<FeedbackResponseAttributes> getViewableFeedbackResponsesForQuestionInSection(
             FeedbackQuestionAttributes question, String userEmail,
-            UserRole role, String section) {
+            UserRole role, String section, SectionDisplayMode sectionDisplayMode) {
 
         List<FeedbackResponseAttributes> viewableResponses = new ArrayList<>();
 
-        // Add responses that the user submitted himself
-        addNewResponses(
-                viewableResponses,
-                getFeedbackResponsesFromGiverForQuestionInSection(
-                        question.getId(), userEmail, section));
-
-        // Add responses that user is a receiver of when question is visible to
-        // receiver.
-        if (question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER)) {
+        switch (role) {
+        case STUDENT:
+            // Add responses that the user submitted himself
             addNewResponses(
                     viewableResponses,
-                    getFeedbackResponsesForReceiverForQuestionInSection(
+                    getFeedbackResponsesFromGiverForQuestionInSection(
                             question.getId(), userEmail, section));
-        }
 
-        switch (role) {
-        case STUDENT:
+            // Add responses that user is a receiver of when question is visible to
+            // receiver.
+            if (question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER)) {
+                addNewResponses(
+                        viewableResponses,
+                        getFeedbackResponsesForReceiverForQuestionInSection(
+                                question.getId(), userEmail, section));
+            }
```

Comment:
Not sure if I get the logic here. Why is this block (from `// Add responses that the user submitted himself` to this line) moved to inside `case STUDENT`?

Reply:
Because all those responses will be retrieved with other responses in line 212. I've also excluded responses not visible for the current instructor.